### PR TITLE
(MODULES-9343) Add Puppet Strings docs

### PR DESCRIPTION
**Pull request by eimlav**

### CODE REVIEW

File: lib/puppet/type/dsc.rb [hunk 1]
```diff
@@ -58,7 +69,8 @@ def change_to_s(currentvalue, newvalue)
   end
 
   newparam(:module) do
-    desc 'DSC Resource Module'
+    desc 'Name of the DSC Resource module to use. For example, the xPSDesiredStateConfiguration DSC Resource module contains '\
+         'the xRemoteFile DSC Resource.'
```
Comment (tphoney) [inline lib/puppet/type/dsc.rb]: this looks odd

File: lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb [hunk 2]
```diff
@@ -103,6 +119,12 @@ def alive?
           self.class.stream_valid?(@stderr)
       end
 
+      # Executes the given PS code.
+      #
+      # @param [???] powershell_code
```
Comment (tphoney) [inline lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb]: ???

File: lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb [hunk 3]
```diff
@@ -138,13 +161,22 @@ def exit
...
       def self.init_path
         # a PowerShell -File compatible path to bootstrap the instance
         path = File.expand_path('../../dsc_lite/templates', __FILE__)
         path = File.join(path, 'init_ps.ps1').tr('/', '\\')
         "\"#{path}\""
       end
 
+      # Generates the given PS code.
+      #
+      # @param [???] powershell_code
```
Comment (tphoney) [inline lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb]: ???

File: lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb [hunk 4]
```diff
@@ -215,7 +246,6 @@ def self.read_length_prefixed_string(bytes)
       # bytes is a binary string containing a list of length-prefixed
       # key / value pairs (of UTF-8 encoded strings)
       # this method mutates the incoming value
-      private_class_method
```
Comment (tphoney) [inline lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb]: are we sure we want to remove them ?

File: lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb [hunk 5]
```diff
@@ -225,9 +255,11 @@ def self.ps_output_to_hash(bytes)
         hash
       end
 
-      # 1 byte command identifier
-      #     0 - Exit
-      #     1 - Execute
+      # Returns byte command identifier for exit and execute commands
+      #
+      # @param [Symbol] command Either :exit or :execute
```
Comment (tphoney) [inline lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb]: add in 0 is :exit 1 is :execute

File: lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb (hunk 4 above)
Comment (eimlav) [reply lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb]: These do not appear to actually do anything. The correct syntax is either:
`private_class_method def self.my_method`
or
```
def self.my_method
...
end
private_class_method :my_method
```

When I attempt to actually make these private there are a number of errors as they are used by public instance methods which can be seen [here](https://ci.appveyor.com/project/puppetlabs/puppetlabs-dsc-lite/builds/25751202/job/8nw63ksk43e7oamt) for example.

File: lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb (hunk 2 above)
Comment (eimlav) [reply lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb]: updated.

File: lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb (hunk 3 above)
Comment (eimlav) [reply lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb]: updated.

File: lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb (hunk 5 above)
Comment (eimlav) [reply lib/puppet_x/puppetlabs/dsc_lite/powershell_manager.rb]: updated.

**Pull request merged**